pandas/tests/io/sas/test_sas.py

```diff
@@ -20,7 +20,15 @@ def test_sas_buffer_format(self):
 
     def test_sas_read_no_format_or_extension(self):
         # see gh-24548
-        msg = "unable to infer format of SAS file"
+        msg = "unable to infer format of SAS file.+"
         with tm.ensure_clean("test_file_no_extension") as path:
             with pytest.raises(ValueError, match=msg):
                 read_sas(path)
+
+
+def test_sas_archive(datapath):
+    fname_uncompressed = datapath("io", "sas", "data", "airline.sas7bdat")
+    df_uncompressed = read_sas(fname_uncompressed)
+    fname_compressed = datapath("io", "sas", "data", "airline.sas7bdat.gz")
```

Comment:
It is worth adding a gzip'd file?  Could the gzip file be created on the fly during the test?  This would also let you test other supported formats, e.g., zip

Reply:
We do have `.gz` files in the repo so I thought that's the way to go. Can also gzip the file in the test. Let me know what's your preference.

Reply:
@bashtage

Reply:
I don't think it is essential to use this method - I just wonder about the longer term future since it isn't strictly necessary to add compressed versions of files to the main repo.  I'm happy to sign off on it as it is now.

Reply:
My 2c. I would prefer these files to by gzipped on the fly in the test such that files in the main repo are absolutely necessary. Doesn't have to be for this PR but a followup would be appreciated